File: ios/ReactNativeViewTests/ReactNativeViewTests.swift
//
//  ReactNativeViewTests.swift
//  ReactNativeViewTests
//
//  Copyright © 2017 Liberty Mutual. All rights reserved.
//

import XCTest
import AppPlatform

class ReactNativeViewTests: XCTestCase {

    override func setUp() {
        super.setUp()
        // Put setup code here. This method is called before the invocation of each test method in the class.
    }

    override func tearDown() {
        // Put teardown code here. This method is called after the invocation of each test method in the class.
        super.tearDown()
    }

    func testInit() {
        if let rnv = try? ReactNativeView(config: ViewConfig(attributes: [
            "name": "ReactNative",
            "config": [
                "bundleURL": "http://localhost:8081/index.ios.bundle?platform=ios",
                "moduleName": "HelloWorld",
                "initialProperties": [AnyHashable("test"): 123],
                "launchOptions": [AnyHashable("something"): "myConfig"]
                ] as [String: Any]
            ]
        )) {
            XCTAssertNotNil(rnv.viewController)
        } else {
            XCTFail("ReactNativeView not initialized")
        }

    }

}

File: android/reactnativeview/src/test/java/com/lmig/uscm/digital/appplatform/reactnativeview/ReactNativeViewTests.kt
package com.lmig.uscm.digital.appplatform.reactnativeview

import org.junit.After
import org.junit.Before
import org.junit.Test
import kotlin.test.assertTrue

/**
 * ReactNativeView Tests
 */
class ReactNativeViewTests {

    private var rnv: ReactNativeView? = null

    @Before
    fun setUp() {
        rnv = ReactNativeView()
    }

    @After
    fun tearDown() {
    }

    @Test
    fun testInit() {
        assertTrue(rnv is ReactNativeView)
    }
}